Bump fuels to 0.50.1

DefiCake opened the pull request.

Braqzen commented on packages/fungible-token/bridge-fungible-token/tests/functions/bridge/mod.rs:
```diff
@@ -338,6 +372,8 @@ mod success {
             .find(|&r| matches!(r, Receipt::MessageOut { .. }))
             .unwrap();
 
+        dbg!(hex::encode(message_receipt.data().unwrap()));
+
```
```suggestion
```

Braqzen commented on packages/fungible-token/bridge-fungible-token/tests/utils/builder.rs:
```diff
@@ -41,35 +48,20 @@ pub async fn build_contract_message_tx(
 
     // Build a change output for the owner of the first provided coin input
     if !gas_coins.is_empty() {
```
No more outputs being added below?

DefiCake replied:
No asset change generated in these file, so yeah, no outputs

Braqzen commented on packages/message-predicates/contract-message-predicate/tests/utils/builder.rs:
```diff
@@ -53,23 +53,79 @@ pub async fn build_contract_message_tx(
+}
+
+/// Build a message-to-contract transaction with the given input coins and outputs, but invalid script bytecode
+/// note: unspent gas is returned to the owner of the first given gas input
```
I guess this is why the outputs were removed from above

Braqzen commented on packages/message-predicates/contract-message-predicate/tests/utils/builder.rs:
```diff
@@ -53,23 +53,79 @@ pub async fn build_contract_message_tx(
+
+/// Build a message-to-contract transaction with the given input coins and outputs, but invalid script bytecode
+/// note: unspent gas is returned to the owner of the first given gas input
+/// Build a message-to-contract transaction with the given input coins and outputs
```
Which message should remain here?

DefiCake replied:
Solved with the correct one

Braqzen reviewed (commented):
Is the hex crate still needed?

DefiCake commented:
> Is the hex crate still needed?

I removed in from the fungible-token-tests, it is still used in message-predicates. I moved it to the root of the project and marked it to be used within the workspace, because I use it here and there when I need to debug
